File: web/src/api/article.ts
import request from "../utils/request";

export const GetAllArticle = () => request({
  url: '/article/list',
  method: 'get',
})

export const GetArticlesById = (id: number) => request({
  url: '/article/list/id',
  method: 'get',
  params: {
    id
  }
})

export const GetArticleByAuthor = () => request({
  url: '/article/author',
  method: 'get',
})

export const GetOneArticle = (id: number) => request({
  url: '/article/get',
  method: 'get',
  params: {
    id
  }
})

export const AddArticle = (title: string, content: string) => request({
  url: '/article/add',
  method: 'post',
  data: {
    title,
    content,
  }
})

export const DeleteArticle = (id: number) => request({
  url: '/article/delete',
  method: 'delete',
  params: {
    id
  }
})

export const UpdateArticle = (id: number, title: string, content: string) => request({
  url: '/article/update',
  method: 'post',
  data: {
    id,
    title,
    content,
  }
})

export const ArticleCount = () => request({
  url: '/article/count',
  method: 'get'
})

File: server/router/article.go
package router

import (
	"github.com/gin-gonic/gin"
	"server/api"
)

func ArticleRouter(router *gin.Engine) {
	articleApi := api.ArticleApi{}

	article := router.Group("api/article")
	{
		article.POST("/add", articleApi.AddArticle)
		article.DELETE("/delete", articleApi.DeleteArticle)
		article.GET("/get", articleApi.GetOneArticle)
		article.GET("/list", articleApi.GetAllArticle)
		article.GET("/list/id", articleApi.GetArticlesById)
		article.GET("/author", articleApi.GetArticleByAuthor)
		article.POST("/update", articleApi.UpdateArticle)
		article.GET("/count", articleApi.ArticleCount)
	}

}
